### DISCUSSION TITLE
Sceneupdate with line, undefined line segments will appear

### DISCUSSION THREAD

**Discussion by SoleyRanTage** [foxglove/community#833] [Q&A] [answered]
Hi guys, when I try to add lines in Sceneupdate, some undefined line segments will appear. I cannot find these points in raw messages either. Did I forget do some definitions or do something wrong? Below are main parts of my code and the screenshot showing the problems. If has any suggestions, please comment, thx!

                   `objects_line = objects_entity->add_lines(); `
                    objects_line->set_type(foxglove::LinePrimitive_Type_LINE_LOOP);
                    objects_line->set_thickness(0.1);
                    objects_line->set_scale_invariant(false);
                    color = objects_line->mutable_color();
                    color->set_r(0);
                    color->set_g(0);
                    color->set_b(255);
                    color->set_a(50);
                    orientation = objects_line->mutable_pose()->mutable_orientation();
                    orientation->set_x(0);
                    orientation->set_y(0);
                    orientation->set_z(0);
                    orientation->set_w(0);
                    position = objects_line->mutable_pose()->mutable_position();
                    position->set_x(0);
                    position->set_y(0);
                    position->set_z(0);
                    for(int i = 0; i < MAX_OBJECT_CONTOUR_POINTS_SIZE; ++i){
                    if(object.contour_points[i][0] <= 0.5)
                        break;
                    objects_points = objects_line->add_points();
                    objects_points->set_x(object.contour_points[i][1]);
                    objects_points->set_y(object.contour_points[i][2]);
                    objects_points->set_z(object.contour_points[i][3]);
                    }`  
![image](https://github.com/foxglove/community/assets/145534575/dc6e57c2-abd2-41c5-a9df-c24fba4b5329)

**Comment by achim-k** [comment]
Could you provide a mcap or bag file with the SceneUpdate topic? That would help a lot in debugging the problem.

**Comment by snosenzo** [comment, marked as answer]
what version of foxglove are you using? If you're not using 1.71.0 (latest version), can you update and see if you can still reproduce?
A similar bug was fixed last week: https://github.com/foxglove/studio/issues/6781

**Reply by SoleyRanTage** [reply]
Yeah, that works after updating! And also when I changed to use line strip in old version, the lines disappeared. Thanks a lot. ^0^
